lib/modules/datasource/crate/index.ts

```diff
@@ -319,31 +319,17 @@ export class CrateDatasource extends Datasource {
           { clonePath, registryFetchUrl },
           `Cloning private cargo registry`,
         );
-
-        const git = Git({
-          ...simpleGitConfig(),
-          maxConcurrentProcesses: 1,
-        }).env(getChildEnv());
-        const clonePromise = git.clone(registryFetchUrl, clonePath, {
-          '--depth': 1,
-        });
+        const clonePromise = CrateDatasource.clone(
+          registryFetchUrl,
+          clonePath,
+          packageName,
+          cacheKeyForError,
+        );
 
         memCache.set(
           cacheKey,
           clonePromise.then(() => clonePath).catch(() => null),
         );
-
-        try {
-          await clonePromise;
-        } catch (err) {
-          logger.warn(
-            { err, packageName, registryFetchUrl },
-            'failed cloning git registry',
-          );
-          memCache.set(cacheKeyForError, err);
-
-          return null;
-        }
       }
```

Comment:
```suggestion
        await clonePromise;
      }
```

we need to wait for the clone to finish